chainerrl/agents/ddpg.py

```diff
@@ -344,6 +344,96 @@ def act(self, obs):
                           self.t, action.array[0], q.array)
         return cuda.to_cpu(action.array[0])
 
+    def batch_act(self, batch_obs):
+        """Select a batch of actions for evaluation.
+
+        Args:
+            batch_obs (Sequence of ~object): Observations.
+
+        Returns:
+            Sequence of ~object: Actions.
+        """
+
+        with chainer.using_config('train', False), chainer.no_backprop_mode():
+            batch_xs = self.batch_states(batch_obs, self.xp, self.phi)
+            batch_action = self.policy(batch_xs).sample()
+            # Q is not needed here, but log it just for information
+            q = self.q_function(batch_xs, batch_action)
+
+        # Update stats
+        self.average_q *= self.average_q_decay
+        self.average_q += (1 - self.average_q_decay) * float(
+            q.array.mean(axis=0))
+        self.logger.debug('t:%s a:%s q:%s',
+                          self.t, batch_action.array[0], q.array)
+        return [cuda.to_cpu(action.array) for action in batch_action]
+
+    def batch_act_and_train(self, batch_obs):
+        """Select a batch of actions for training.
+
+        Args:
+            batch_obs (Sequence of ~object): Observations.
+
+        Returns:
+            Sequence of ~object: Actions.
+        """
+
+        batch_greedy_action = self.batch_act(batch_obs)
+        batch_action = [
+            self.explorer.select_action(
+                self.t, lambda: batch_greedy_action[i])
+            for i in range(len(batch_greedy_action))]
+        self.t += 1
```

Comment:
This line should be removed since `self.t` is incremented in `batch_observe_and_train`.